src/main/java/org/gridsuite/geodata/server/GeoDataService.java

```diff
@@ -125,6 +130,59 @@ List<SubstationGeoData> getSubstations(Network network, Set<Country> countries)
         return new ArrayList<>(substationsGeoData.values());
     }
 
+    List<SubstationGeoData> getSubstations(Network network, List<String> substationsIds) {
+        String escapedIds = StringUtils.join(substationsIds.stream().map(LogUtils::sanitizeParam).collect(Collectors.toList()), ", ");
+        LOGGER.info("Loading substations geo data for substations with ids {} of network '{}'", escapedIds, network.getId());
+
+        StopWatch stopWatch = StopWatch.createStarted();
```

Comment:
You don't add this here ?
`Objects.requireNonNull(network);`

Reply:
Done